### PR TITLE
Code cleanups and bug fixes

### PR DESCRIPTION
**Pull request by we-ai**
This PR:
- Implements code cleanups described in issue [#871](https://github.com/episphere/connect/issues/871)
- Improves query in BQ
- Improves error handling when retrieving notification schemas

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -1960,21 +1953,15 @@ const retrieveNotificationSchemaByID = async (id) => {
...
-    if ((getDrafts && docData.isDraft) || (!getDrafts && !docData.isDraft)) result.push(docData);
+  let query = db.collection("notificationSpecifications").where("isDraft", "==", getDrafts);
+  if (category !== "all") {
+    query = query.where("category", "==", category);
+  } else {
+    query = query.orderBy("category");
   }
 
-  return result;
+  const snapshot = await query.orderBy("attempt").get();
```
Comment (anthonypetersen) [inline utils/firestore.js:1963]: will this `orderBy` override the one in the previous `else` statement? or does it act as an additional `orderBy` similar to how a SQL query would work?

File: utils/notifications.js [hunk 2]
```diff
@@ -110,18 +110,26 @@ const markAllNotificationsAsAlreadyRead = (notification, collection) => {
 }
 
 const retrieveNotifications = async (req, res, uid) => {
+  if (req.method !== "GET") {
+    return res.status(405).json(getResponseJSON("Only GET requests are accepted!", 405));
```
Comment (anthonypetersen) [inline utils/notifications.js:114]: unrelated - but thoughts on creating a shared function that can be utilized for common non-200 statuses such as 405 and 500? We could ensure consist messaging and formatting of these responses.

**Review by anthonypetersen — APPROVED** [review]
looks good, couple of random comments made that don't affect approval status

File: utils/firestore.js (hunk 1 above)
Comment (we-ai) [reply utils/firestore.js:1963]: I believe it works similarly to SQL's way. [Here](https://github.com/firebase/snippets-web/blob/e74e8fd9591f70007a44c58580ff0f1fcd1c7959/firestore/test.firestore.js#L942-L942) is an example used in its documentation.

File: utils/notifications.js (hunk 2 above)
Comment (we-ai) [reply utils/notifications.js:114]: Agree. We should discuss and do it.

Comment (anthonypetersen) [reply utils/notifications.js:114]: was a new index required? if so, we will want to remember when code goes to each environment

File: utils/firestore.js (hunk 1 above)
Comment (anthonypetersen) [reply utils/firestore.js:1963]: was a new index required? if so, we will want to remember when code goes to each environment

Comment (we-ai) [reply utils/firestore.js:1963]: Yes. Created dev and stage indexes already. Need your help creating a prod index.

**Review by jhflorey — APPROVED** [review]

**Pull request merged**